### PR TITLE
New getting started

### PR DESCRIPTION
**Pull request by dunglas**
WIP

### CODE REVIEW

File: tutorial/api.md [hunk 1]
```diff
@@ -1,276 +1,122 @@
...
+included in the Standard Edition. This Doctrine ORM bridge is optimized for performance and development convenience. Doctrine
+ORM and its bridge support major RDBMS including MySQL, PostgreSQL, SQLite, SQL Server and MariaDB.
 
-Mapping your app data model to Schema.org structures can be a tedious task. Using the generator, your data model will be
-a derived from Schema.org. Serializing your data as JSON-LD will not require specific mapping nor adaptation. It's a matter
-of minutes.
+Open `http://localhost` with your favorite web browser:
 
-**Ready for the future**
+![Swagger UI integration in API Platform](images/swagger-ui.png)
```
Comment (meyerbaptiste) [inline tutorial/api.md]: `images/swagger-ui-1.png`

File: tutorial/api.md [hunk 2]
```diff
@@ -1,276 +1,122 @@
...
-and process data from any website or app using such technologies.
+API Platform exposes a description of the API in the Swagger UI format. It also integrates Swagger UI, a nice interface
+rendering the API documentation. Click on an operation to display its details. You can also send requests to the API directly
+from the UI. Try to create a new foo object using the `POST` operation, then access it using `GET` and finally  delete it
+with `DELETE`.
+If you access any API URL using a web browser, API Platform will detect it (using the `Accept` HTTP header send by the header)
+and display will the related request to the API in the UI. Open `http://localhost/foos`:
 
-Ask Doctrine to create the database of the project:
+![Request detail in the UI](images/swagger-ui.png)
```
Comment (meyerbaptiste) [inline tutorial/api.md]: `images/swagger-ui-2.png`

File: tutorial/api.md [hunk 3]
```diff
@@ -1,276 +1,122 @@
...
-    $ bin/console doctrine:database:create
+If you want to access the raw data, you have two alternatives:
 
-Then generate database tables related to the generated entities:
+* Add the correct `Accept` header (or don't set any `Accept` header at all and API Platform will default to JSON-LD) - preferred
+  when writing API clients
+* Add the format format you want as the extension of the resource - to debug purpose only
 
-    $ bin/console doctrine:schema:create
+For instance, go to `http://localhost/foos.jsonld` to retrieve the list of Foos object in JSON-LD or http://localhost/foos.json`
```
Comment (meyerbaptiste) [inline tutorial/api.md]: Missing backtick before `http://localhost/foos.json`

File: tutorial/api.md [hunk 4]
```diff
@@ -1,276 +1,122 @@
...
+4. Add some validation rules
+
+API Platform will use the data model to expose a read/write web API having a ton of built-in features:
+
+* data validation
+* pagination
+* filtering
+* sorting
+* a nice UI and machine-readable documentations ([Swagger/OpenAPI](https://swagger.io), [Hydra](http://hydra-cg.com))
+* hypermedia/HATEOAS and content negotiation support ([JSON-LD](http://json-ld.org), [HAL](blog.stateless.co/post/13296666138/json-linking-with-hal))
```
Comment (meyerbaptiste) [inline tutorial/api.md]: [`http://blog.stateless.co/post/13296666138/json-linking-with-hal`](blog.stateless.co/post/13296666138/json-linking-with-hal)

File: tutorial/api.md [hunk 5]
```diff
@@ -1,276 +1,122 @@
...
+this operation is done only one time. Starting servers will then be lightning fast.
 
-To scaffold our blog data model we'll browse [Schema.org](https://schema.org) and find types matching our needs.
-We're lucky, the [https://schema.org/BlogPosting](https://schema.org/BlogPosting) describes exactly the data model we want
-for our blog. As you can see, there is a bunch of schemas available.
+Project's files are automatically shared between your local machine and the container thanks to a pre-configured [Docker
+volume](https://docs.docker.com/engine/tutorials/dockervolumes/). It means that you can edit files of your project locally
+using your preferred IDE or code editor, they will be transparently taken into account.
+Speaking about IDEs, our preferred software to develop API Platform apps is [PHPStorm](https://www.jetbrains.com/phpstorm/)
+with its awesome "Symfony" and "PHP annotations" plugins. Give it a try, you will got auto-completion for almost everything.
```
Comment (meyerbaptiste) [inline tutorial/api.md]: Maybe add a link to https://plugins.jetbrains.com/plugin/7219 and https://plugins.jetbrains.com/plugin/7320?

File: tutorial/api.md (hunk 4 above)
Comment (meyerbaptiste) [inline tutorial/api.md]: Maybe add a link to https://en.wikipedia.org/wiki/HATEOAS?

File: tutorial/api.md [hunk 6]
```diff
@@ -1,276 +1,122 @@
...
+environment up and running. The bundled setup contains an image pre-configured with PHP 7, Apache and everything required
+to run API Platform and a MySQL image to host the database.
 
-When you're done with the demo app and want to create your own API:
+*Alternatively to using Docker, API Platform can also be installed using [Composer](https://getcomposer.org/):
+`composer create-project api-platform/api-platform bookshop-api`.*
 
-* empty the `app/config/schema.yml` file
-* delete all PHP files in the `src/AppBundle/Entity/` directory
+Start by [downloading the API Platform Standard Edition](https://api-platform.com/download) and extract the content of the
```
Comment (meyerbaptiste) [inline tutorial/api.md]: There is a plan to update this page? As naive user, when I am on this page I see `composer create-project api-platform/api-platform my-project` or a link to https://github.com/api-platform/api-platform but no archive. It's confusing.

Comment (dunglas) [inline tutorial/api.md]: Yes! I'll do it when the new doc will be ready.

**Comment by dunglas** [comment]
Ping @api-platform/core-team, the main new tutorial is ready to be reviewed.

File: tutorial/api.md [hunk 7]
```diff
@@ -1,608 +1,613 @@
...
+is compatible with most [Symfony bundles](https://symfony.com/blog/the-30-most-useful-symfony-bundles-and-making-them-even-better)
+(plugins) and benefits from the numerous extensions points provided by this rock-solid foundation (events, DIC...).
+Adding features like custom, service-oriented, API endpoints, JWT or OAuth authentication, HTTP caching, mail sending or
+asynchronous jobs to your APIs is very straightforward.
 
 ## Installing the framework
 
-Let's start our new blog API project. The easiest way to create a new project is to use [Composer](https://getcomposer.org/)
-(you need to have it installed on your box):
+API Platform is shipped with a entire [Docker](https://docker.com) setup that makes it easy to get a containerized development
```
Comment (soyuka) [inline tutorial/api.md]: with a ~~entire~~ Docker setup

File: tutorial/api.md [hunk 8]
```diff
@@ -1,608 +1,613 @@
...
+environment up and running. This setup contains an image pre-configured with PHP 7, Apache and everything needed to run API
```
Comment (soyuka) [inline tutorial/api.md]: btw stack could be simplified with php-fpm (nothing to do with the getting-started guide though)

File: tutorial/api.md [hunk 9]
```diff
@@ -1,608 +1,613 @@
...
+Platform and a MySQL image to host the database.
 
-    composer create-project api-platform/api-platform:dev-master blog-api
+*Alternatively to using Docker, API Platform can also be installed using [Composer](https://getcomposer.org/):
+`composer create-project api-platform/api-platform bookshop-api`.*
```
Comment (soyuka) [inline tutorial/api.md]: As an alternative to Docker, API Platform can also be installed using [Composer](https://getcomposer.org/):

```
composer create-project api-platform/api-platform bookshop-api
```

Why emphasis?

File: tutorial/api.md [hunk 10]
```diff
@@ -1,608 +1,613 @@
...
 
-Composer creates the skeleton of the new blog API then retrieve the framework and all its dependencies.
+Start by [downloading the API Platform Standard Edition](https://api-platform.com/download) and extract the content of the
```
Comment (soyuka) [inline tutorial/api.md]: ...and extract it's content.

File: tutorial/api.md [hunk 11]
```diff
@@ -1,608 +1,613 @@
...
+archive.
+The resulting directory contains an empty API Platform project structure. You will add your own code and configuration inside
```
Comment (soyuka) [inline tutorial/api.md]: I'm not sure this sentence is needed as we repeat "go inside the directory containing your project skeleton" below

File: tutorial/api.md [hunk 12]
```diff
@@ -1,608 +1,613 @@
...
+it.
+Then, if you do not already have Docker on your computer, [it's the right time to install it](https://www.docker.com/products/overview#/install_the_platform).
 
-At the end of the installation, you will be prompted for some configuration parameters including database credentials.
-All configuration parameters can be changed later by editing the `app/config/parameters.yml` file.
+Open a terminal, go inside the directory containing your project skeleton, and run the following command to start the Apache
+and the MySQL servers using [Docker Compose](https://docs.docker.com/compose/):
```
Comment (soyuka) [inline tutorial/api.md]: Open a terminal, and navigate to the directory containing your project skeleton. Then, run the following command to start the Apache-Php-Mysql stack: 

```
docker-compose up
```

`-d` for daemon?

File: tutorial/api.md [hunk 13]
```diff
@@ -1,608 +1,613 @@
...
 
-API Platform is pre-configured to use the popular and powerful [Doctrine ORM](http://www.doctrine-project.org/projects/orm.html).
-It's supported natively by all API Platform components. However the Doctrine ORM is fully optional: you can replace it
-by your favorite ORM, no ORM at all and even no database.
+    $ docker-compose up
 
-The installer will also ask you for some configuration parameters:
+The first time you start the containers, Docker downloads and builds images for you. It will take some time, but don't worry,
+this operation is done only one time. Starting servers will then be lightning fast.
```
Comment (soyuka) [inline tutorial/api.md]: this is done only once.

File: tutorial/api.md [hunk 14]
```diff
@@ -1,608 +1,613 @@
...
 
-* `database_*`: database credentials (MySQL is configured by default, but other popular RDBMS are supported)
-* `mailer_*`: mail server credentials (to send mails)
-* `cors_allow_origin`: the URL of your default web client application to automatically set appropriate [CORS](http://en.wikipedia.org/wiki/Cross-origin_resource_sharing)
-  headers, **set it to `http://locahost:9000` (the default URL of the built-in Grunt server of our AngularJS client) to follow this tutorial**
-* `api_*`: a name and a description of the API that will be used in the generated documentation
-* `secret`: a secret token (choose a long one) for cryptographic features
+Project's files are automatically shared between your local host machine and the container thanks to a pre-configured [Docker
+volume](https://docs.docker.com/engine/tutorials/dockervolumes/). It means that you can edit files of your project locally
+using your preferred IDE or code editor, they will be transparently taken into account in the container.
```
Comment (soyuka) [inline tutorial/api.md]: ~~ preferred~~ favorite

File: tutorial/api.md [hunk 15]
```diff
@@ -1,608 +1,613 @@
...
+Speaking about IDEs, our preferred software to develop API Platform apps is [PHPStorm](https://www.jetbrains.com/phpstorm/)
+with its awesome [Symfony](https://confluence.jetbrains.com/display/PhpStorm/Getting+Started+-+Symfony+Development+using+PhpStorm)
+and [PHP annotations](https://plugins.jetbrains.com/plugin/7320) plugins. Give them a try, you'll got auto-completion for
+almost everything.
 
-Finally, the installer will ask if you want to remove the existing VCS history, type `Y`.
+Now, in another shell, install the PHP dependencies of the project:
```
Comment (soyuka) [inline tutorial/api.md]: Now, in another shell, install the project's PHP dependencies:

btw If it's daemonized we only need one shell.

File: tutorial/api.md [hunk 16]
```diff
@@ -1,608 +1,613 @@
...
 
-Take a look at [the content of the generated directory](https://github.com/dunglas/blog-api). You maybe recognize a [Symfony
-application directory structure](https://symfony.com/doc/current/quick_tour/the_architecture.html). It's fine and intended:
-**the generated skeleton is a perfectly valid Symfony full-stack application** that follows [Symfony Best Practices](https://symfony.com/doc/current/best_practices/index.html).
+    $ docker-compose run web composer install --no-interaction
 
-It means that with this default API Platform setup, you can:
+The `web` container is where your project belongs. Prefixing a command by `docker-compose run web` allows to execute this
```
Comment (soyuka) [inline tutorial/api.md]: The `web` container is where your project stands. Prefixing a command by `docker-compose run web` allows to execute the given command in the container. You may want [to create an alias](http://www.linfo.org/alias.html) to easily run commands inside the container. Here, we are installing libraries required by the project using the `composer` tool included in the API Platform image.

File: tutorial/api.md [hunk 17]
```diff
@@ -1,608 +1,613 @@
...
+in the container. Here, we installed libraries required by the project using the `composer` tool included in the API Platform
+image.
 
-* [use thousands of exiting Symfony bundles](http://knpbundles.com)
-* reuse all your Symfony skills and benefit of the high quality [Symfony documentation](https://symfony.com/doc/current/index.html)
+The API Platform Standard Edition comes with a dummy entity for test purpose: `src/AppBundle/Entity/Foo.php`. We will remove
+it later, but for now, create the related database table:
 
-You can also use API Platform in any existing Symfony application.
+    docker-compose run web bin/console doctrine:schema:create
```
Comment (soyuka) [inline tutorial/api.md]: Sometimes you use ` sometimes identation for code?

File: tutorial/api.md [hunk 18]
```diff
@@ -1,608 +1,613 @@
...
+If you're used to the PHP ecosystem, you probably guessed that this test entity uses the industry leading [Doctrine ORM](http://www.doctrine-project.org/projects/orm.html)
+library as persistence system.
+API Platform is 100% independent of the persistence system and you can use the one(s) that best suit(s) your needs (like
+a NoSQL database or a remote web service).
+API Platform even supports using several persistence systems together in the same project.
 
-The skeleton comes with a demonstration bookstore API. You can play with it by running the following commands:
+However, Doctrine ORM is definitely the easiest way to persist and query data in an API Platform project thanks to a bridge
+included in the Standard Edition. This Doctrine ORM bridge is optimized for performance and development convenience. Doctrine
+ORM and its bridge support major RDBMS including MySQL, PostgreSQL, SQLite, SQL Server and MariaDB.
```
Comment (soyuka) [inline tutorial/api.md]: supports

File: tutorial/api.md [hunk 19]
```diff
@@ -1,608 +1,613 @@
...
+     *
+     * @ORM\ManyToOne(targetEntity="Book", inversedBy="reviews")
+     */
+    private $book;
+}
+```
 
-Reusing an existing semantic schema like we've done has many advantages:
+As you can see there are just 2 typical PHP objects with the corresponding PHPDoc (you will see that the PHPDoc of entities
+and properties will appear in the API documentation).
```
Comment (soyuka) [inline tutorial/api.md]: As you can see those are two typical PHP objects with the corresponding PHPDoc (note that entities PHPDoc
 and their properties will appear in the API documentation).

File: tutorial/api.md [hunk 20]
```diff
@@ -1,608 +1,613 @@
...
 
-When you create new entities, you need to clear the cache:
+By default, the API supports `GET` (retrieve, on collections and items), `POST` (create), `PUT` (update) and `DELETE` (self-explaining)
+HTTP methods. You are not limited to the built-in operations. You can [add new custom operations](../core/operations.md#enabling-and-disabling-operations)
+(`PATCH` operations, sub-resources...) or [disable the ones you don't want](../core/operations.md#creating-custom-operations-and-controllers).
 
-    $ bin/console cache:clear
+Try the `GET` operation on the collection. The book we added appears. When the collection will contain more than 30 items,
+the pagination will automatically show up, [and this is entirely configurable](../core/pagination.md). You may be interested
+in [adding some filters and in sorting the collection](../core/filters.md) as well.
```
Comment (soyuka) [inline tutorial/api.md]: adding some filters or adding sorts to the collection.

File: tutorial/api.md [hunk 21]
```diff
@@ -1,608 +1,613 @@
...
 
-Start the integrated development web server:
+Maybe did you notice that some keys start by the `@` symbol in the generated JSON response (`@id`, `@type`, `@context`...)?
```
Comment (soyuka) [inline tutorial/api.md]: You may have notice that some keys start with the `@` symbol...

File: tutorial/api.md [hunk 22]
```diff
@@ -1,608 +1,613 @@
...
+API Platform comes with a full support of the [JSON-LD](http://json-ld.org/) format (and its [Hydra](http://www.hydra-cg.com/)
+extension). It allows to build smart clients, with auto-discoverability capabilities (take a look at [Hydra console](http://www.markus-lanthaler.com/hydra/console/))
+and is very useful for open data, SEO and interoperability when [used with open vocabularies such as Schema.org](http://blog.schema.org/2013/06/schemaorg-and-json-ld.html).
+JSON-LD enables a lot of awesome advanced features (like [giving access to Google to your structured data](https://developers.google.com/search/docs/guides/intro-structured-data)
+or consuming APIs with [Apache Jena](https://jena.apache.org/documentation/io/#formats)).
+Wwe think that it's the best default for a new API. However, API Platform natively [supports many other formats](../core/content-negotiation.md)
```
Comment (soyuka) [inline tutorial/api.md]: We think that it's the best default format for a new API. However, API Platform natively [supports many other formats](../core/content-negotiation.md)

File: tutorial/api.md [hunk 23]
```diff
@@ -1,608 +1,613 @@
...
+}
+```
 
-![API doc](images/api-doc.png)]
+Be sure to replace `<THE-UUID-OF-THE-BOOK>` by... The UUID of the book we previously created.
 
-Thanks to [NelmioApiDocBundle](https://github.com/nelmio/NelmioApiDocBundle) support of ApiBundle and its integration
-with API Platform, you get for a free **an automatically generated human-readable documentation** of the API (Swagger-like).
-The doc also **includes a sandbox** to try the API.
+There is 2 interesting things to mention about this request:
```
Comment (soyuka) [inline tutorial/api.md]: There are

File: tutorial/api.md [hunk 24]
```diff
@@ -1,608 +1,613 @@
...
 
-You can also use your favorite HTTP client (yeah, we already talked about Postman) to query the API.
-It is lower level than the sandbox and will allow to inspect forge and inspect JSON requests and responses easily.
+First, we learned how to work with relations. In a hypermedia API, every resource is identified by an (unique) [IRI](https://en.wikipedia.org/wiki/Internationalized_Resource_Identifier).
+An URL is a valid IRI, and it's what API Platform uses. The `@id` property of every JSON-LD document contains the IRI identifying
+it. You can use this IRI to reference this document from other documents. In the previous request, we used the IRI of the
+book we created earlier to link it with the `Review` we were creating. API Platform is smart enough to deal with IRIs.
+By the way, you may want to [embed documents](../core/serialization-groups-and-relations.md) instead of referencing them (e.g. to reduce the number of HTTP requests
+to issue).
```
Comment (soyuka) [inline tutorial/api.md]: (e.g. to reduce the number of HTTP requests ~~to issue~~).

File: tutorial/api.md [hunk 25]
```diff
@@ -1,608 +1,613 @@
...
-  "birthDate": "0666-06-06T00:00:00+00:00",
-  "deathDate": null,
-  "description": "A famous author from the North.",
-  "familyName": "Lenancker",
-  "givenName": "Olivier"
-}
-```
+Did you notice that the error was automatically serialized in JSON-LD and respect the Hydra Core vocabulary for errors?
+It allows the client to easily extract useful information from the error. Anyway, it's bad to get a SQL error when submitting
+a request. It means that we doesn't valid use input, and [it's a very bad and dangerous practice](https://www.owasp.org/index.php/Input_Validation_Cheat_Sheet).
```
Comment (soyuka) [inline tutorial/api.md]: It means that we doesn't use a valid input,

File: tutorial/api.md [hunk 26]
```diff
@@ -1,608 +1,613 @@
...
-      "name": "API Platform is great",
-      "kevinReview": "nice"
-    }
-    """
-```
-
-The API Platform flavor of Behat also comes with a temporary SQLite database dedicated to tests. It works out of the box.
-
-Simply run `vendor/bin/behat`. Everything should be green:
+After updating the entities to add those `@Assert\*` annotations (as for Doctrine, you can use XML or YAML formats if you
```
Comment (soyuka) [inline tutorial/api.md]: by adding those `@Assert\*` annotations

File: core/getting-started.md [hunk 27]
```diff
@@ -38,7 +38,7 @@ api:
     prefix:   '/api' # Optional
 ```
 
-There is no mandatory configuration. However, a lot of [configuration parameters](configuration.md) are available.
+There is no mandatory configuration. However, a bunch of [configuration parameters](configuration.md) are available.
```
Comment (theofidry) [inline core/getting-started.md]: There is no mandatory configuration although [some parameters](configuration.md) are configurable.

File: tutorial/api.md [hunk 28]
```diff
@@ -1,608 +1,613 @@
...
-in mind that you can use your preferred client-side technology (tested and approved with Angular, React, Ionic, Swift but
-can work with any language able to send HTTP requests).
-
-## Prerequisites
-
-PHP 7 must be installed to run API Platform. A built-in web server is shipped with the framework for the development environment.
-
-To follow this tutorial a database must be installed (but its not a strong dependency of the framework). We recommend MySQL
-or MariaDB but other major DBMS are supported including PostgreSQL and SQLite.
+[API Platform](https://api-platform.com) is the most efficient framework out there to create web APIs. It makes it
```
Comment (theofidry) [inline tutorial/api.md]: is the most efficient modern PHP framework to create web APIs.

File: tutorial/api.md (hunk 8 above)
Comment (dunglas) [inline tutorial/api.md]: Yes, PR welcome!

File: tutorial/api.md (hunk 9 above)
Comment (dunglas) [inline tutorial/api.md]: It's italic.

File: tutorial/api.md (hunk 11 above)
Comment (dunglas) [inline tutorial/api.md]: IMO It's less clear without it.

File: tutorial/api.md (hunk 17 above)
Comment (dunglas) [inline tutorial/api.md]: It depends if it is on a single line or not (and if it's PHP or shell commands).

File: tutorial/api.md (hunk 28 above)
Comment (dunglas) [inline tutorial/api.md]: It changes the meaning of the sentence ;)

Comment (dunglas) [inline tutorial/api.md]: It's a marketing sentence. Someone can argue that Apigility or FOSRest or anything else is more efficient. Someone can also argue that Django REST or Loopback is more efficient. It's just subjective marketing :)

File: tutorial/api.md (hunk 12 above)
Comment (soyuka) [inline tutorial/api.md]: to start the Apache and Mysql?

File: tutorial/api.md (hunk 9 above)
Comment (soyuka) [inline tutorial/api.md]: Why italic ? :P

File: tutorial/api.md (hunk 8 above)
Comment (soyuka) [inline tutorial/api.md]: Ok, if we remove apache php-fpm will listen on an upstream (like `localhost:8080`). Is it sufficient or you'd prefer having `localhost` (eg nginx proxy on top).

File: tutorial/api.md [hunk 29]
```diff
@@ -1,608 +1,612 @@
...
+easy to create simple APIs with support of industry leading open standards while giving you the ability to create complex
```
Comment (teohhanhui) [inline tutorial/api.md]: ~~simple~~ APIs

This gives the wrong idea.

Comment (teohhanhui) [inline tutorial/api.md]: with the support of

File: tutorial/api.md [hunk 30]
```diff
@@ -1,608 +1,612 @@
...
+features with style. To discover the basics, we will create an API to manage a bookshop.
```
Comment (teohhanhui) [inline tutorial/api.md]: `create complex features with style`?

I'm not sure what this part of the sentence is trying to say...

Comment (teohhanhui) [inline tutorial/api.md]: Perhaps we should rephrase this sentence to something like this:

> It makes it easy to start creating APIs with the support of industry-leading open standards, while giving you the flexibility to build complex features.

File: tutorial/api.md [hunk 31]
```diff
@@ -1,608 +1,612 @@
...
+
+API Platform uses these model classes to expose a web API having a ton of built-in features:
+
+* creating, retrieving, updating and deleting (CRUD) resources
+* data validation
+* pagination
+* filtering
+* sorting
+* content negotiation
+* a nice UI and machine-readable documentations ([Swagger/OpenAPI](https://swagger.io), [Hydra](http://hydra-cg.com))
```
Comment (teohhanhui) [inline tutorial/api.md]: documentation *(uncountable)*

File: tutorial/api.md [hunk 32]
```diff
@@ -1,608 +1,612 @@
...
-* reuse all your Symfony skills and benefit of the high quality [Symfony documentation](https://symfony.com/doc/current/index.html)
+The API Platform Standard Edition comes with a dummy entity for test purpose: `src/AppBundle/Entity/Foo.php`. We will remove
+it later, but for now, create the related database table:
 
-You can also use API Platform in any existing Symfony application.
+    docker-compose run web bin/console doctrine:schema:create
 
-While API Platform is perfectly integrated with Symfony, keep in mind that it's basically a set of standalone PHP components.
-You can also use those components in raw PHP (without framework) as well as with other frameworks.
+If you're used to the PHP ecosystem, you probably guessed that this test entity uses the industry leading [Doctrine ORM](http://www.doctrine-project.org/projects/orm.html)
```
Comment (teohhanhui) [inline tutorial/api.md]: industry-leading

File: tutorial/api.md (hunk 28 above)
Comment (dunglas) [inline tutorial/api.md]: Can you suggest an alternative wording? What do you think about "API Platform is one the most efficient framework out there to create web APIs."?

Comment (theofidry) [inline tutorial/api.md]: We can keep the "best in the world" or "best framework out there" part in the homepage, let's be more neutral for the doc :P

Comment (teohhanhui) [inline tutorial/api.md]: > API Platform is a framework for creating modern web APIs efficiently.

**Comment by dunglas** [comment]
This PR is ready now. Can you review please?

**Pull request merged**